plugins/modules/nxos_fc_interfaces.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2023 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for nxos_fc_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: nxos_fc_interfaces
+short_description: Fc Interfaces resource module
+description: This module manages the interface attributes of NX-OS fc interfaces.
+version_added: 1.0.0
+author: Suhas Bharadwaj (@srbharadwaj)
+notes:
+  - Tested against NXOS 9.3(2) on Cisco MDS Switches
+options:
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the NX-OS
+        device by executing the command B(show running-config interface)
+      - The state I(parsed) reads the configuration from C(running_config)
+        option and transforms it into Ansible structured data as per the
+        resource module's argspec and the value is then returned in the
+        I(parsed) key within the result.
+    type: str
+  config:
+    description: A dictionary of interface options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Full name of interface, e.g. fc1/1, fc18/48
+        type: str
+        required: true
+      description:
+        description:
+          - Interface description.
+        type: str
+      enabled:
+        description:
+          - Administrative state of the interface. Set the value to C(true) to
+            administratively enable the interface or C(false) to disable it
+        type: bool
+      speed:
+        description:
+          - Interface link speed.
+        choices:
+          - auto
+          - 1000
+          - 2000
+          - 4000
+          - 8000
+          - 10000
+          - 16000
+          - 32000
+          - 64000
+          - auto max 2000
+          - auto max 4000
+          - auto max 8000
+          - auto max 16000
+          - auto max 32000
+          - auto max 64000
+        type: str
+      mode:
+        description:
+          - Port mode of the fc interface
+        choices:
+          - auto
+          - E
+          - F
+          - Fx
+          - NP
+          - SD
+        type: str
+      trunk_mode:
+        description:
+          - Trunk mode of the fc interface
+        choices:
+          - auto
+          - on
+          - off
+        type: str
+      analytics:
+        description:
+          - Analytics type on the fc interface
+        choices:
+          - fc-scsi
+          - fc-nvme
+          - fc-all
+        type: str
+  state:
+    description:
+      - The state of the configuration after module completion
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - gathered
+      - rendered
+      - parsed
+    default: merged
+"""
+EXAMPLES = """
```

Comment:
Please add some examples in the following format:

https://github.com/ansible-collections/cisco.nxos/blob/main/plugins/modules/nxos_interfaces.py#L128